gef.py

```diff
@@ -11085,18 +11094,25 @@ def connect(self, pid: int) -> bool:
 
     def setup(self) -> bool:
         # setup remote adequately depending on remote or qemu mode
+        success = True
+
         if self.in_qemu_user():
             dbg(f"Setting up as qemu session, target={self.__qemu}")
-            self.__setup_qemu()
+            if not self.__setup_qemu():
+                success = False
         else:
             dbg(f"Setting up as remote session")
-            self.__setup_remote()
+            if not self.__setup_remote():
+                success = False
 
         # refresh gef to consider the binary
         reset_all_caches()
-        gef.binary = Elf(self.lfile)
+
+        if self.file.exists():
+            gef.binary = Elf(self.lfile)
```

Comment:
This is problematic as it will prevent any function/command depending on `gef.binary` for instance `checksec`) to work at all. Are you doing this check because of the support for `rr`? If so, we probably still need the way I mentioned to detect we're running `rr` so that we can gracefully handle that case. 
I'm not super sure the best way about this, but that `if` is definitely problematic.

Reply:
This is not related to rr, actually this has never failed in my tests, but raising an exception in the setup of the session would mean that the session would be running (since it is actually managed by gdb itself) but the gef session's wrapper would be completely broken.

I think having some commands not working (like checksec) is better than having nothing working.

But I agree with you that something needs to be done to prevent having a "bad" fail when running those commands, and instead having a clean error message